src/tests/suite/daos_test_common.c

```diff
@@ -814,53 +814,63 @@ run_daos_sub_tests(char *test_name, const struct CMUnitTest *tests,
 	return rc;
 }
 
-void
-daos_exclude_target(const uuid_t pool_uuid, const char *grp,
-		    const d_rank_list_t *svc, d_rank_t rank,
-		    int tgt_idx)
+static void
+daos_dmg_pool_target(const char *sub_cmd, const uuid_t pool_uuid,
+		    const char *grp, const char *dmg_config,
+		    const d_rank_list_t *svc, d_rank_t rank, int tgt_idx)
 {
-	struct d_tgt_list	targets;
-	int			rc;
+	char		dmg_cmd[DTS_CFG_MAX];
+	char		dmg_options[DTS_CFG_MAX] = {0};
+	int		rc;
 
-	/** exclude from the pool */
-	targets.tl_nr = 1;
-	targets.tl_ranks = &rank;
-	targets.tl_tgts = &tgt_idx;
-	rc = daos_pool_tgt_exclude(pool_uuid, grp, svc, &targets, NULL);
-	if (rc)
-		print_message("exclude pool failed rc %d\n", rc);
+	/* build and invoke dmg cmd */
+	if (tgt_idx != -1)
+		dts_create_config(dmg_options, " --target-idx=%d", tgt_idx);
+	if (dmg_config != NULL)
```

Comment:
is this mutually exclusive from tgt_idx because this command overwrites the previous one ?  Should it be an else if ?  Or should we append?

Reply:
Good point, that goes to the previous comment above, if dts_create_config overwrites dmg_options on each call, then, this is an issue, they should be appended. I will work on this.